gradle-plugin/src/main/kotlin/io/reflekt/plugin/ReflektSubPlugin.kt

```diff
@@ -8,35 +10,65 @@ import java.io.File
 import org.gradle.api.provider.Provider
 
 @Suppress("unused")
 class ReflektSubPlugin :  KotlinCompilerPluginSupportPlugin {
+    private val reflektMetaFile = "ReflektMeta"
+    private val metaInfDir = "META-INF"
 
     override fun applyToCompilation(kotlinCompilation: KotlinCompilation<*>): Provider<List<SubpluginOption>> {
         println("ReflektSubPlugin loaded")
         val project = kotlinCompilation.target.project
         val extension = project.reflekt
 
-        val filesToIntrospect: MutableSet<File> = HashSet()
+        val reflektMetaFiles: MutableSet<File> = HashSet()
         project.configurations.forEach {
-            filesToIntrospect.addAll(getFilesToIntrospect(getJarFilesToIntrospect(it, extension)))
+            reflektMetaFiles.addAll(getReflektMetaFiles(getJarFilesToIntrospect(it, extension)))
         }
-        val librariesToIntrospect = filesToIntrospect.map { SubpluginOption(key = INTROSPECT_FILE_OPTION_INFO.name, value = it.absolutePath) }
+        val reflektMetaFilesOptions = reflektMetaFiles.map { SubpluginOption(key = REFLEKT_META_FILE_OPTION_INFO.name, value = it.absolutePath) }
         val dependencyJars = project.configurations.first { it.name == "compileClasspath" }
             .map { SubpluginOption(key = DEPENDENCY_JAR_OPTION_INFO.name, value = it.absolutePath) }
 
+        val librariesToIntrospect = extension.librariesToIntrospect.map { SubpluginOption(key = LIBRARY_TO_INTROSPECT.name, value = it) }
+        val generationPath = "${project.buildDir.absolutePath}/${extension.generationPath}"
+
+        with(project) {
+            afterEvaluate {
+                project.mySourceSets.apply {
+                    this.getAt("main").kotlin.srcDir(generationPath)
+                }
+            }
+        }
+
         return project.provider {
-            librariesToIntrospect + dependencyJars +
+            librariesToIntrospect + reflektMetaFilesOptions + dependencyJars +
                 SubpluginOption(key = ENABLED_OPTION_INFO.name, value = extension.enabled.toString()) +
-                SubpluginOption(key = OUTPUT_DIR_OPTION_INFO.name, value = extension.generationPath)
+                SubpluginOption(key = OUTPUT_DIR_OPTION_INFO.name, value = generationPath) +
+                SubpluginOption(key = SAVE_METADATA_OPTION_INFO.name, value = extension.toSaveMetadata.toString()) +
+                SubpluginOption(key = REFLEKT_META_FILE_PATH.name, value = createReflektMeta(project.getResourcesPath()).absolutePath)
         }
     }
 
+    private fun createReflektMeta(resourcesDir: String): File {
+        val metaInfDir = File("$resourcesDir/${metaInfDir}")
+        if (!metaInfDir.exists()) {
+            metaInfDir.mkdirs()
+        }
+        return File("${metaInfDir.path}/${reflektMetaFile}")
+    }
+
+    // TODO: can we do it better?
+    private fun Project.getResourcesPath(): String = "${project.rootDir}${project.path.replace(":", "/")}/src/main/resources"
```

Comment:
maybe this would work (but cannot test, unfortunately)
`project.mySourceSets.getAt("main").resources.first().absolutePath`

Reply:
I am not sure, why you take only the first resource? In Any case it will be on another PR